src/core/cancellation_middleware.py

```
@@ -24,16 +24,20 @@ async def on_call_tool(self, context: MiddlewareContext, call_next):
         
         try:
-            logger.debug(f"Starting tool call: {getattr(context, 'method', 'unknown')}")
+            tool_method = getattr(context, 'method', 'unknown')
+            logger.info(f"🛡️  CANCELLATION MIDDLEWARE: Starting tool call: {tool_method}")
+            logger.info(f"🔍 Context type: {type(context).__name__}")
+            
             result = await call_next(context)
             
             duration_ms = (time.perf_counter() - start_time) * 1000
-            logger.debug(f"Tool call completed in {duration_ms:.2f}ms")
+            logger.info(f"✅ CANCELLATION MIDDLEWARE: Tool call completed successfully in {duration_ms:.2f}ms")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)